pynest/nest/tests/test_stdp_synapse.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+#
+# test_stdp_synapse.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import nest
+import unittest
+from math import exp
+import numpy as np
+
+try:
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+    DEBUG_PLOTS = True
+except Exception:
+    DEBUG_PLOTS = False
+
+
+@nest.ll_api.check_stack
+class STDPSynapseTest(unittest.TestCase):
+    """
+    Compare the STDP synaptic plasticity model against a self-contained Python reference.
+
+    Random pre and post spike times are generated according to a Poisson distribution; some hard-coded spike times are
+    added to make sure to test for edge cases such as simultaneous pre- and post spike.
+    """
+
+    def init_params(self):
+        self.resolution = 0.1    # [ms]
+        self.simulation_duration = 1E3    # [ms]
+        self.synapse_model = "stdp_synapse"
+        self.presynaptic_firing_rate = 20.    # [ms^-1]
+        self.postsynaptic_firing_rate = 20.    # [ms^-1]
+        self.tau_pre = 16.8
+        self.tau_post = 33.7
+        self.init_weight = .5
+        self.synapse_parameters = {
+            "synapse_model": self.synapse_model,
+            "receptor_type": 0,
+            "delay": self.dendritic_delay,
+            # STDP constants
+            "lambda": 0.01,
+            "alpha": 0.85,
+            "mu_plus": 0.0,
+            "mu_minus": 0.0,
+            "tau_plus": self.tau_pre,
+            "Wmax": 15.0,
+            "weight": self.init_weight
+        }
+        self.neuron_parameters = {
+            "tau_minus": self.tau_post,
+        }
+        self.hardcoded_trains_length = 15
```

Comment:
Is this the sum of the pre- and post-lengths? If yes, why not compute it from the arrays. Otherwise, comment.